### PR TITLE
BCDA-8105 Add cclf-import terraform infrastructure 

### PR DESCRIPTION
**Pull request by austincanada**
## 🎫 Ticket

https://jira.cms.gov/browse/BCDA-8105

## 🛠 Changes

Adding Terraform infrastructure for cclf-import lambda functions.
 
## ℹ️ Context for reviewers

Used existing opt-out import service as a template.
cclf-import lambda will only be used by BCDA.
This is partly a check of work. 

## ✅ Acceptance Validation

Not verified. If things look good from a code sense, testing can happen. Not sure on testing and how to perform it.

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

File: .github/workflows/cclf-import-apply.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./actions/setup-tfenv-terraform
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::${{ matrix.app == 'bcda' && secrets[format('{0}_{1}_ACCOUNT', matrix.app, matrix.env)] || secrets.BCDA_ACCOUNT }}:role/delegatedadmin/developer/${{ matrix.app }}-${{ matrix.env }}-github-actions
+          aws-region: ${{ vars.AWS_REGION }}
+      - run: terraform init -reconfigure -backend-config=../../backends/$TF_VAR_app-$TF_VAR_env.s3.tfbackend
+      - run: terraform apply -auto-approve
+
```
Comment (gsf) [inline .github/workflows/cclf-import-apply.yml:44]: Please add a cclf-import-plan workflow as well.

**Review by gsf — CHANGES_REQUESTED** [review]

File: .github/workflows/cclf-import-apply.yml (hunk 1 above)
Comment (gsf) [reply .github/workflows/cclf-import-apply.yml:44]: Once the plan workflow is added we can use that to see the terraform plan output for verification.

File: terraform/services/cclf-import/main.tf [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  memory_size = local.memory_size[var.app]
+
+  function_role_inline_policies = {
+    assume-bucket-role = data.aws_iam_policy_document.assume_bucket_role.json
+  }
+
+  environment_variables = {
+    ENV      = var.env
+    APP_NAME = "${var.app}-${var.env}-cclf-import"
+    DB_HOST  = data.aws_ssm_parameter.cclf_db_host.value #TODO: What is this
```
[the remark this answers is not in the document]
Comment (austincanada) [reply terraform/services/cclf-import/main.tf]: Nope, those are stale. Removed.

File: terraform/services/cclf-import/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# TODO: Are these needed? Any extras?
```
[the remark this answers is not in the document]
Comment (austincanada) [reply terraform/services/cclf-import/outputs.tf]: Stale TODO, removed.

File: .github/workflows/cclf-import-apply.yml (hunk 1 above)
Comment (austincanada) [reply .github/workflows/cclf-import-apply.yml:44]: Plan workflow added. Is running the plan workflow something that should happen locally or during another type of check?

Comment (oluwolenpbc) [reply .github/workflows/cclf-import-apply.yml:44]: idk about locally but the plan should trigger sucessfully before the apply triggers for sure.

File: terraform/services/cclf-import/main.tf [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+data "aws_iam_policy_document" "assume_bucket_role" {
+  statement {
+    actions   = ["sts:AssumeRole"]
+    resources = [data.aws_ssm_parameter.bfd_bucket_role_arn.value]
+  }
+}
+
+data "aws_ssm_parameter" "cclf_db_host" {
+  name = "/${var.app}/${var.env}/cclf/db-host"
+}
```
Comment (kyeah) [inline terraform/services/cclf-import/main.tf:29]: We can remove this, the app code won't use this :)

File: .github/workflows/cclf-import-apply.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      id-token: write
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./terraform/services/cclf-import
+    strategy:
+      fail-fast: false
+      matrix:
+        app: [bcda]
+        env: [dev, test, prod]
```
Comment (kyeah) [inline .github/workflows/cclf-import-apply.yml:31]: Hmm we'll need to import CCLF files in opensbx as well, but I realized something while adding values to parameter store. Will follow up in slack 🤔

Comment (kyeah) [reply .github/workflows/cclf-import-apply.yml:31]: Actually, maybe I'm overthinking this. Opensbx probably doesn't need this lambda at this point in time 😄

File: .github/workflows/cclf-import-apply.yml (hunk 1 above)
Comment (austincanada) [reply .github/workflows/cclf-import-apply.yml:44]: Should be good to go now.

File: terraform/services/cclf-import/main.tf (hunk 4 above)
Comment (austincanada) [reply terraform/services/cclf-import/main.tf]: Removed.

File: .github/workflows/cclf-import-apply.yml (hunk 5 above)
Comment (austincanada) [reply .github/workflows/cclf-import-apply.yml:31]: Sounds good, plan looks like it's running correctly!

**Review by kyeah — APPROVED** [review]

**Review by gsf — APPROVED** [review]
Plan looks good! Let's merge and see how the apply runs

File: terraform/services/cclf-import/main.tf (hunk 4 above)
Comment (gsf) [reply terraform/services/cclf-import/main.tf]: Quick check - does the app code still access the DB in some way?

**Pull request merged**